src/index.ts

```diff
@@ -63,7 +63,9 @@ function createRdfEmitter(program: Program) {
 
         // CLASS PART
         // Checks if Model is actual class (model Truck) or Model is a data property (model CUSIP is string)
-        if (m.properties.size != 0) {
+        const intrinsicName = getIntrinsicModelName(program, m);

```

Comment:
The real thing you intend to test here is "is this a scalar type or a model type", and this is the recommended way to do so for now. The code you had does work but might fail in the case where a user wants an object type that happens to not have any properties.